Riot/Modules/MatrixKit/Utils/EventFormatter/MarkdownToHTMLRenderer.swift

```diff
@@ -50,3 +57,103 @@ public class MarkdownToHTMLRendererHardBreaks: MarkdownToHTMLRenderer {
     }
     
 }
+
+// MARK: - AST-handling private extensions
+private extension CMarkNode {
+    /// Formatting symbol associated with given note type
+    /// Note: this is only defined for node types that are handled in repairLinks
+    var formattingSymbol: String {
+        switch self.type {
+        case CMARK_NODE_EMPH:
+            return "_"
+        case CMARK_NODE_STRONG:
+            return "__"
+        default:
+            return ""
+        }
+    }
+
+    /// Repairs links that were broken down by markdown formatting.
+    /// Should be used on the first node of libcmark's AST
+    /// (e.g. the object returned by DownASTRenderer.stringToAST).
+    func repairLinks() {
```

Comment:
Not going to pretend I understand everything this does but I trust that we'll find out pretty fast if it breaks anything 😅

Reply:
Yeah, I already added a nice coverage with unit tests. But if someone ever finds a string that manages to break it, I suppose we will just have to fix it and extend our unit tests with it :)
If you have suggestions of weird strings that I could add to unit tests, feel free